### ISSUE TITLE
Watch not triggering

### ISSUE THREAD

**Issue by bori87** [TeamMaestro/angular-native-seed#46]
Compilation complete. Watching for file changes.

Nothing happening on file changes.

**Comment by gofreddo** [comment]
+1

**Comment by fourctv** [comment]
Is that on nativescript? On {N} you need to open a separate terminal and run `npm run livesync`.

If not {N}, can you describe your environment? It works fine here (OS X, Chrome).

**Comment by bori87** [comment]
Yes is on {N} . I dont think gulp watch follows symlinks. I am on windows 10.

**Comment by fourctv** [comment]
Have you tried to start a separate cmd-line session and run `npm run livesync` there?
You need that because {N} is monitoring changes to the `app` folder, and you are making changes to `src`, so the modified files need to be copied to `app`. That is what the `livesync` script does, monitor `src` and copy changes over to `app`.

**Comment by bori87** [comment]
Yes i have two cmd windows open.  gulp.watch([`../${SRC}/**/*`]) is working but gulp.watch([`${SRC}/**/*`]) is not.

**Comment by bori87** [comment]
I modified the gulp file and now i am watchig my web folder. Gulp watch is not working on symlinks subfolders.

Below code working for me:
 return gulp.watch([`../${SRC}/**/*`])
        .on('change', (file) => {
            var outputDest = file.substring(0, file.lastIndexOf('\\')+1).replace(SRC.substring(0, SRC.length - 1), DEST).replace('..\\', '');

**Comment by sean-perkins** [comment]
We do not use Windows machines, so it's very likely the way that symlinks are handled in OS X vs. Windows are different. 

I would suggest using the options block with `{ follow: true }` over modifying the watch directory.

**Comment by sergiocosus** [comment]
I have the same issue with Ubuntu 16.04

**Comment by Tompote** [comment]
I didn't see this issue before posting so I close mine but i ran into the same problem.
Livesync seems to work with 3.9.1 but not with 4.0-alpha-2 (https://github.com/TeamMaestro/angular-native-seed/issues/51).

**Comment by sean-perkins** [comment]
@Tompote thanks for investigating this issue further and debugging. I would honestly prefer keeping the gulp version at 3.9.x, until I can diagnose what underlying change breaks symlinks in 4.0-x. The goal of the project structure is to allow multiple integration points and easily re-use scripts without much hassle or manipulation.

This would allow more simplistic integrations such as electron or leveraging Vue.js instead of Angular. 

I'm just getting back from vacation this afternoon, so I'll confirm the gulp version conflict and update master within the next few days.

**Comment by db3dev** [comment]
This project will not work with versions of gulp before version 4. Please submit any PRs on this issue for Gulp 4.

**Comment by ITServicesAM** [comment]
no progress here? 
I can not use the repo for my app, because there is no livesync for nativescript at all.

**Comment by sean-perkins** [comment]
@ITServicesAM That statement is just not true. On a Mac the livesync works as expected. If you have a windows, feel free to adjust the script and problem solve why gulp sync doesn't recognize the changes.

**Comment by harrijon** [comment]
I love the idea of this seed and sharing codebases between native and web.  It seems as though it is really close to being awesome.  ..except this livesync issue on windows 10.

@sean-perkins What is the exact workaround for windows 10?

I have tried editing nativescript/gulpfile.js, per suggestions above from @Tompote  and @bori87 , but, to no avail.  ..I must not be doing it correctly, so, detailed instructions or a permanent fix would be greatly appreciated.

**Comment by sean-perkins** [comment]
We do not use Windows machines, so we have no recommended workaround. If the community is using Windows machines, the community needs to submit a working PR around this issue.

Related issues:
https://github.com/gulpjs/gulp/issues/222
https://github.com/gulpjs/gulp/issues/1491

The project requires Gulp 4, so the previous recommendations are not applicable.

**Comment by bori87** [comment]
This code is working for me:
gulpfile.js
```javascript
gulp.task('tns.Livesync', () => {
    return gulp.watch([`../${SRC}**/*.common.ts`, `../${SRC}**/*.tns.ts`, `../${SRC}**/*.tns.html`,  `../${SRC}**/*.service.ts`,
     `../${SRC}**/*.tns.scss`, `../${SRC}**/*.scss`, `../${SRC}**/*.component.ts`, `../${SRC}**/*.routes.ts`,
     `../${SRC}**/*.index.ts`])
        .on('change', (file) => {
            var outputDest = file.substring(0, file.lastIndexOf('\\')+1).replace(SRC.substring(0, SRC.length - 1), DEST).replace('..\\', '');
            gulp.src([file])
                .pipe(rename(removeTns))
                .pipe(replace('.scss\'', '.css\'', { logs: { enabled: true }}))
                .pipe(debug({title: 'tns.Livesync'}))
                .pipe(debug({title: 'out ' + outputDest}))
                .pipe(gulp.dest(outputDest, {overwrite: true}));
        });
});
```

**Comment by harrijon** [comment]
Thx @bori87, that worked!

In case any other windows 10 users need it:

In addition to @bori87 's gulpfile.js changes, 2 cmd windows need to be open to use nativescript livesync
-1st cmd window:  In the "nativescript" folder:  `npm run android`
-2nd cmd window:  In the "nativescript" folder:  `npm run livesync`


Thx @sean-perkins for adding "help wanted" tag

**Comment by ITServicesAM** [comment]
Hi folks,

my statement above is a little bit over the top. This repo is awesome, it has the best mix of included features and is not overloaded so it is easy to understand and to start using it.

For me it was a little bit frustrating to not be able to use it on a windows machine.

@bori87 thx for the solution it is working as expected.

**Comment by MahendraTennakoon** [comment]
I'm experiencing a similar issue on Windows. I've updated the gulpfile.js as @bori87 has mentioned, the watch triggers and the console prints "successfully synced application", but the latest changes do not get applied to the app.

**Comment by muthukumaroptisol** [comment]
I am getting the same issue on Windows 10. Gulp file has been modified as per @bori87 mentioned and seems like watch does work as expected, but the changes are not updated in app folder. @bori87 can you look into his ?

**Comment by simano65** [comment]
See: https://stackoverflow.com/questions/49866461/angular-native-seed-live-sync-is-not-working-in-windows-10/49875909#49875909

**Comment by muthukumaroptisol** [comment]
Below code works for me windows 10

gulp.task('tns.Livesync', () => {
    return gulp.watch([`../${SRC}**/*.common.ts`, `../${SRC}**/*.tns.ts`, `../${SRC}**/*.tns.html`,  `../${SRC}**/*.service.ts`,
     `../${SRC}**/*.tns.scss`, `../${SRC}**/*.scss`, `../${SRC}**/*.component.ts`, `../${SRC}**/*.routes.ts`,
     `../${SRC}**/*.index.ts`])
        .on('change', (file) => {
            var outputDest = file.substring(0, file.lastIndexOf('\\')+1).replace('src\\', 'app\\').replace('..\\', '');
            gulp.src([file])
                .pipe(rename((path) => { path.dirname = ''; }))
                .pipe(rename(removeTns))
                .pipe(replace('.scss\'', '.css\'', { logs: { enabled: true }}))
                .pipe(debug({title: 'tns.Livesync'}))
                .pipe(debug({title: 'out ' + outputDest}))
                .pipe(gulp.dest(outputDest, {overwrite: true}));
        });
});
